File: app/src/main/java/com/varma/hemanshu/coffeehouse/MainActivity.java
package com.varma.hemanshu.coffeehouse;

import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    int quantityCap = 0;
    int quantityMocha = 0;
    int quantityFrappe = 0;
    int quantityEspresso = 0;
    String summaryString;
    String name;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    /**
     * This method is called when increment button is clicked.
     */
    public void increment(View view) {
        Button incButtonCappuccino = findViewById(R.id.IncButtonViewCappuccino);
        Button incButtonMocha = findViewById(R.id.IncButtonViewMocha);
        Button incButtonFrappe = findViewById(R.id.IncButtonViewFrappe);
        Button incButtonEspresso = findViewById(R.id.IncButtonViewEspresso);
        switch (view.getId()) {
            case R.id.IncButtonViewCappuccino:
                if (quantityCap == 100) {
                    incButtonCappuccino.setEnabled(false);
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityCap += 1;
                }
                incButtonCappuccino.setEnabled(true);
                displayQuantityCap(quantityCap);
                break;
            case R.id.IncButtonViewMocha:
                if (quantityMocha == 100) {
                    incButtonMocha.setEnabled(false);
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityMocha += 1;
                }
                incButtonMocha.setEnabled(true);
                displayQuantityMocha(quantityMocha);
                break;
            case R.id.IncButtonViewFrappe:
                if (quantityFrappe == 100) {
                    incButtonFrappe.setEnabled(false);
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityFrappe += 1;
                }
                incButtonFrappe.setEnabled(true);
                displayQuantityFrappe(quantityFrappe);
                break;
            case R.id.IncButtonViewEspresso:
                if (quantityEspresso == 100) {
                    incButtonEspresso.setEnabled(false);
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityEspresso += 1;
                }
                incButtonEspresso.setEnabled(true);
                displayQuantityEspresso(quantityEspresso);
                break;
        }

    }

    /**
     * This method is called when decrement button is clicked.
     */
    public void decrement(View view) {
        Button decButtonCappuccino = findViewById(R.id.decButtonViewCappuccino);
        Button decButtonMocha = findViewById(R.id.decButtonViewMocha);
        Button decButtonFrappe = findViewById(R.id.decButtonViewFrappe);
        Button decButtonEspresso = findViewById(R.id.decButtonViewEspresso);
        switch (view.getId()) {
            case R.id.decButtonViewCappuccino:
                if (quantityCap == 0) {
                    decButtonCappuccino.setEnabled(false);
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityCap -= 1;
                }
                decButtonCappuccino.setEnabled(true);
                displayQuantityCap(quantityCap);
                break;
            case R.id.decButtonViewMocha:
                if (quantityMocha == 0) {
                    decButtonMocha.setEnabled(false);
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityMocha -= 1;
                }
                decButtonMocha.setEnabled(true);
                displayQuantityMocha(quantityMocha);
                break;
            case R.id.decButtonViewFrappe:
                if (quantityFrappe == 0) {
                    decButtonFrappe.setEnabled(false);
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityFrappe -= 1;
                }
                decButtonFrappe.setEnabled(true);
                displayQuantityFrappe(quantityFrappe);
                break;
            case R.id.decButtonViewEspresso:
                if (quantityEspresso == 0) {
                    decButtonEspresso.setEnabled(false);
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show();
                } else {
                    quantityEspresso -= 1;
                }
                decButtonEspresso.setEnabled(true);
                displayQuantityEspresso(quantityEspresso);
                break;
        }
    }

    /**
     * This method displays the given quantity value of CAPPUCCINO on screen.
     */
    private void displayQuantityCap(int number) {
        TextView quantityTextViewCap = findViewById(R.id.quantity_text_view_cap);
        quantityTextViewCap.setText("" + number);

    }

    /**
     * This method displays the given quantity value of CAFFE MOCHA on screen.
     */
    private void displayQuantityMocha(int number) {
        TextView quantityTextViewMocha = findViewById(R.id.quantity_text_view_mocha);
        quantityTextViewMocha.setText("" + number);

    }

    /**
     * This method displays the given quantity value of FRAPPE on screen.
     */
    private void displayQuantityFrappe(int number) {
        TextView quantityTextViewFrappe = findViewById(R.id.quantity_text_view_frappe);
        quantityTextViewFrappe.setText("" + number);

    }

    /**
     * This method displays the given quantity value of ESPRESSO on screen.
     */
    private void displayQuantityEspresso(int number) {
        TextView quantityTextViewEspresso = findViewById(R.id.quantity_text_view_espresso);
        quantityTextViewEspresso.setText("" + number);

    }

    /**
     * This method is called when VIEW SUMMARY Button is clicked.
     */
    public void viewSummary(View view) {
        int finalPrice;
        // Checks for HotCoffee Extra's cappuccino
        CheckBox addHotCoffeeCap = findViewById(R.id.HotCoffeeCap);
        boolean hasHotCoffeeCap = addHotCoffeeCap.isChecked();
        //Checks for Brewed Coffee Extra's for Cappuccino
        CheckBox addBrewedCoffeeCap = findViewById(R.id.BrewedCap);
        boolean hasBrewedCoffeeCap = addBrewedCoffeeCap.isChecked();

        // Checks for HotCoffee Extra's Mocha
        CheckBox addHotCoffeeMocha = findViewById(R.id.HotCoffeeMocha);
        boolean hasHotCoffeeMocha = addHotCoffeeMocha.isChecked();
        //Checks for Brewed Coffee Extra's for Mocha
        CheckBox addBrewedCoffeeMocha = findViewById(R.id.BrewedMocha);
        boolean hasBrewedCoffeeMocha = addBrewedCoffeeMocha.isChecked();

        // Checks for HotCoffee Extra's Frappe
        CheckBox addHotCoffeeFrappe = findViewById(R.id.HotCoffeeFrappe);
        boolean hasHotCoffeeFrappe = addHotCoffeeFrappe.isChecked();
        //Checks for Brewed Coffee Extra's for frappe
        CheckBox addBrewedCoffeeFrappe = findViewById(R.id.BrewedFrappe);
        boolean hasBrewedCoffeeFrappe = addBrewedCoffeeFrappe.isChecked();

        // Checks for HotCoffee Extra's Espresso
        CheckBox addHotCoffeeEspresso = findViewById(R.id.HotCoffeeEspresso);
        boolean hasHotCoffeeEspresso = addHotCoffeeEspresso.isChecked();
        //Checks for Brewed Coffee Extra's for Espresso
        CheckBox addBrewedCoffeeEspresso = findViewById(R.id.BrewedEspresso);
        boolean hasBrewedCoffeeEspresso = addBrewedCoffeeEspresso.isChecked();

        int priceCap = calculatePriceCap(hasHotCoffeeCap, hasBrewedCoffeeCap);
        int priceMocha = calculatePriceMocha(hasHotCoffeeMocha, hasBrewedCoffeeMocha);
        int priceFrappe = calculatePriceFrappe(hasHotCoffeeFrappe, hasBrewedCoffeeFrappe);
        int priceEspresso = calculatePriceEspresso(hasHotCoffeeEspresso, hasBrewedCoffeeEspresso);
        finalPrice = priceCap + priceMocha + priceFrappe + priceEspresso;

        //Intent for getting name from UserActivity.
        Intent i = getIntent();
        name = i.getStringExtra("Name");
        String phone_no = i.getStringExtra("Phone_No");
        summaryString = createSummary(name, phone_no, hasHotCoffeeCap, hasBrewedCoffeeCap, priceCap, hasHotCoffeeMocha, hasBrewedCoffeeMocha, priceMocha, hasHotCoffeeFrappe, hasBrewedCoffeeFrappe, priceFrappe, hasHotCoffeeEspresso, hasBrewedCoffeeEspresso, priceEspresso, finalPrice);
        displayMessage();


    }

    /**
     * This method displays the Summary in an DialogBox.
     */
    private void displayMessage() {
        AlertDialog.Builder alertDialog = new AlertDialog.Builder(MainActivity.this);
        alertDialog.setTitle(R.string.title);
        alertDialog.setMessage(summaryString);
        alertDialog.setPositiveButton(R.string.okay, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {

            }
        });
        alertDialog.show();
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Cappuccino
     */
    private int calculatePriceCap(boolean isHotCoffee, boolean isBrewedCoffee) {
        int price = 50;
        if (isHotCoffee) {
            price += 5;
        }
        if (isBrewedCoffee) {
            price += 7;
        }
        return quantityCap * price;
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Caffe Mocha
     */
    private int calculatePriceMocha(boolean isHotCoffee, boolean isBrewedCoffee) {
        int price = 70;
        if (isHotCoffee) {
            price += 5;
        }
        if (isBrewedCoffee) {
            price += 7;
        }
        return quantityMocha * price;
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Frappe Coffee
     */
    private int calculatePriceFrappe(boolean isHotCoffee, boolean isBrewedCoffee) {
        int price = 75;
        if (isHotCoffee) {
            price += 5;
        }
        if (isBrewedCoffee) {
            price += 7;
        }
        return quantityFrappe * price;
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Espresso
     */
    private int calculatePriceEspresso(boolean isHotCoffee, boolean isBrewedCoffee) {
        int price = 90;
        if (isHotCoffee) {
            price += 5;
        }
        if (isBrewedCoffee) {
            price += 7;
        }
        return quantityEspresso * price;
    }

    /**
     * This method creates an Summary of Total order by User.
     */
    private String createSummary(String name, String phone, boolean hasHotCoffeeCap, boolean hasBrewedCoffeeCap, int pCap, boolean hasHotCoffeeMocha, boolean hasBrewedCoffeeMocha, int pMocha, boolean hasHotCoffeeFrappe, boolean hasBrewedCoffeeFrappe, int pFrappe, boolean hasHotCoffeeEspresso, boolean hasBrewedCoffeeEspresso, int pEspresso, int price) {

        String summary = getString(R.string.name_summary) + name + "\n" + getString(R.string.phone_no_summary) + phone + "\n" + getString(R.string.view_line);
        //Cappuccino
        summary += "\n" + getString(R.string.cappuccino);
        summary += "\n" + getString(R.string.hot_coffee_summary) + hasHotCoffeeCap + "\n" + getString(R.string.brewed_coffee_summary) + hasBrewedCoffeeCap;
        summary += "\n" + getString(R.string.quantity_summary) + quantityCap + "\n" + getString(R.string.price) + pCap;
        summary += "\n" + getString(R.string.view_line);
        //Mocha
        summary += "\n" + getString(R.string.caffe_mocha);
        summary += "\n" + getString(R.string.hot_coffee_summary) + hasHotCoffeeMocha + "\n" + getString(R.string.brewed_coffee_summary) + hasBrewedCoffeeMocha;
        summary += "\n" + getString(R.string.quantity_summary) + quantityMocha + "\n" + getString(R.string.price) + pMocha;
        summary += "\n" + getString(R.string.view_line);
        //Frappe
        summary += "\n" + getString(R.string.frappe);
        summary += "\n" + getString(R.string.hot_coffee_summary) + hasHotCoffeeFrappe + "\n" + getString(R.string.brewed_coffee_summary) + hasBrewedCoffeeFrappe;
        summary += "\n" + getString(R.string.quantity_summary) + quantityFrappe + "\n" + getString(R.string.price) + pFrappe;
        summary += "\n" + getString(R.string.view_line);
        //Espresso
        summary += "\n" + getString(R.string.espresso);
        summary += "\n" + getString(R.string.hot_coffee_summary) + hasHotCoffeeEspresso + "\n" + getString(R.string.brewed_coffee_summary) + hasBrewedCoffeeEspresso;
        summary += "\n" + getString(R.string.quantity_summary) + quantityEspresso + "\n" + getString(R.string.price) + pEspresso;
        summary += "\n" + getString(R.string.view_line);
        //Total Price
        summary += "\n" + getString(R.string.total_price) + price + "\n" + getString(R.string.view_line);
        return summary;
    }

    /**
     * This method is invoked when CONFIRM ORDER is clicked.
     * It also passes the Summary to Mail using Intent.
     */
    public void mailOrder(View view) {
        String subject = getString(R.string.order_subject) + name;
        Toast.makeText(MainActivity.this, R.string.thank_you, Toast.LENGTH_LONG).show();
        Intent intent = new Intent(Intent.ACTION_SENDTO);
        intent.setData(Uri.parse("mailto:"));
        intent.putExtra(Intent.EXTRA_SUBJECT, subject);
        intent.putExtra(Intent.EXTRA_TEXT, summaryString);
        startActivity(intent);
    }
}

File: app/src/main/java/com/varma/hemanshu/coffeehouse/MainActivity.kt
package com.varma.hemanshu.coffeehouse

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import android.widget.TextView
import android.widget.CheckBox
import android.content.Intent
import android.net.Uri
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {

    private var quantityCap = 0
    private var quantityMocha = 0
    private var quantityFrappe = 0
    private var quantityEspresso = 0
    var summaryString: String? = null
    var name: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    /**
     * This method is called when increment button is clicked.
     */
    fun increment(view: View) {
        val incButtonCappuccino = findViewById<Button>(R.id.IncButtonViewCappuccino)
        val incButtonMocha = findViewById<Button>(R.id.IncButtonViewMocha)
        val incButtonFrappe = findViewById<Button>(R.id.IncButtonViewFrappe)
        val incButtonEspresso = findViewById<Button>(R.id.IncButtonViewEspresso)
        when (view.id) {
            R.id.IncButtonViewCappuccino -> {
                if (quantityCap == 100) {
                    incButtonCappuccino.isEnabled = false
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityCap += 1
                }
                incButtonCappuccino.isEnabled = true
                displayQuantityCap(quantityCap)
            }
            R.id.IncButtonViewMocha -> {
                if (quantityMocha == 100) {
                    incButtonMocha.isEnabled = false
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityMocha += 1
                }
                incButtonMocha.isEnabled = true
                displayQuantityMocha(quantityMocha)
            }
            R.id.IncButtonViewFrappe -> {
                if (quantityFrappe == 100) {
                    incButtonFrappe.isEnabled = false
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityFrappe += 1
                }
                incButtonFrappe.isEnabled = true
                displayQuantityFrappe(quantityFrappe)
            }
            R.id.IncButtonViewEspresso -> {
                if (quantityEspresso == 100) {
                    incButtonEspresso.isEnabled = false
                    Toast.makeText(this, getString(R.string.max_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityEspresso += 1
                }
                incButtonEspresso.isEnabled = true
                displayQuantityEspresso(quantityEspresso)
            }
        }
    }

    /**
     * This method is called when decrement button is clicked.
     */
    fun decrement(view: View) {
        val decButtonCappuccino = findViewById<Button>(R.id.decButtonViewCappuccino)
        val decButtonMocha = findViewById<Button>(R.id.decButtonViewMocha)
        val decButtonFrappe = findViewById<Button>(R.id.decButtonViewFrappe)
        val decButtonEspresso = findViewById<Button>(R.id.decButtonViewEspresso)
        when (view.id) {
            R.id.decButtonViewCappuccino -> {
                if (quantityCap == 0) {
                    decButtonCappuccino.isEnabled = false
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityCap -= 1
                }
                decButtonCappuccino.isEnabled = true
                displayQuantityCap(quantityCap)
            }
            R.id.decButtonViewMocha -> {
                if (quantityMocha == 0) {
                    decButtonMocha.isEnabled = false
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityMocha -= 1
                }
                decButtonMocha.isEnabled = true
                displayQuantityMocha(quantityMocha)
            }
            R.id.decButtonViewFrappe -> {
                if (quantityFrappe == 0) {
                    decButtonFrappe.isEnabled = false
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityFrappe -= 1
                }
                decButtonFrappe.isEnabled = true
                displayQuantityFrappe(quantityFrappe)
            }
            R.id.decButtonViewEspresso -> {
                if (quantityEspresso == 0) {
                    decButtonEspresso.isEnabled = false
                    Toast.makeText(this, getString(R.string.min_quantity), Toast.LENGTH_SHORT).show()
                } else {
                    quantityEspresso -= 1
                }
                decButtonEspresso.isEnabled = true
                displayQuantityEspresso(quantityEspresso)
            }
        }
    }

    /**
     * This method displays the given quantity value of CAPPUCCINO on screen.
     */
    private fun displayQuantityCap(number: Int) {
        val quantityTextViewCap = findViewById<TextView>(R.id.quantity_text_view_cap)
        quantityTextViewCap.text = "$number"
    }

    /**
     * This method displays the given quantity value of CAFFE MOCHA on screen.
     */
    private fun displayQuantityMocha(number: Int) {
        val quantityTextViewMocha = findViewById<TextView>(R.id.quantity_text_view_mocha)
        quantityTextViewMocha.text = "$number"
    }

    /**
     * This method displays the given quantity value of FRAPPE on screen.
     */
    private fun displayQuantityFrappe(number: Int) {
        val quantityTextViewFrappe = findViewById<TextView>(R.id.quantity_text_view_frappe)
        quantityTextViewFrappe.text = "$number"
    }

    /**
     * This method displays the given quantity value of ESPRESSO on screen.
     */
    private fun displayQuantityEspresso(number: Int) {
        val quantityTextViewEspresso = findViewById<TextView>(R.id.quantity_text_view_espresso)
        quantityTextViewEspresso.text = "$number"
    }

    /**
     * This method is called when VIEW SUMMARY Button is clicked.
     */
    fun viewSummary(view: View?) {
        val finalPrice: Int
        // Checks for HotCoffee Extra's cappuccino
        val addHotCoffeeCap = findViewById<CheckBox>(R.id.HotCoffeeCap)
        val hasHotCoffeeCap = addHotCoffeeCap.isChecked
        //Checks for Brewed Coffee Extra's for Cappuccino
        val addBrewedCoffeeCap = findViewById<CheckBox>(R.id.BrewedCap)
        val hasBrewedCoffeeCap = addBrewedCoffeeCap.isChecked

        // Checks for HotCoffee Extra's Mocha
        val addHotCoffeeMocha = findViewById<CheckBox>(R.id.HotCoffeeMocha)
        val hasHotCoffeeMocha = addHotCoffeeMocha.isChecked
        //Checks for Brewed Coffee Extra's for Mocha
        val addBrewedCoffeeMocha = findViewById<CheckBox>(R.id.BrewedMocha)
        val hasBrewedCoffeeMocha = addBrewedCoffeeMocha.isChecked

        // Checks for HotCoffee Extra's Frappe
        val addHotCoffeeFrappe = findViewById<CheckBox>(R.id.HotCoffeeFrappe)
        val hasHotCoffeeFrappe = addHotCoffeeFrappe.isChecked
        //Checks for Brewed Coffee Extra's for frappe
        val addBrewedCoffeeFrappe = findViewById<CheckBox>(R.id.BrewedFrappe)
        val hasBrewedCoffeeFrappe = addBrewedCoffeeFrappe.isChecked

        // Checks for HotCoffee Extra's Espresso
        val addHotCoffeeEspresso = findViewById<CheckBox>(R.id.HotCoffeeEspresso)
        val hasHotCoffeeEspresso = addHotCoffeeEspresso.isChecked
        //Checks for Brewed Coffee Extra's for Espresso
        val addBrewedCoffeeEspresso = findViewById<CheckBox>(R.id.BrewedEspresso)
        val hasBrewedCoffeeEspresso = addBrewedCoffeeEspresso.isChecked
        val priceCap = calculatePriceCap(hasHotCoffeeCap, hasBrewedCoffeeCap)
        val priceMocha = calculatePriceMocha(hasHotCoffeeMocha, hasBrewedCoffeeMocha)
        val priceFrappe = calculatePriceFrappe(hasHotCoffeeFrappe, hasBrewedCoffeeFrappe)
        val priceEspresso = calculatePriceEspresso(hasHotCoffeeEspresso, hasBrewedCoffeeEspresso)
        finalPrice = priceCap + priceMocha + priceFrappe + priceEspresso

        //Intent for getting name from UserActivity.
        val i = intent
        name = i.getStringExtra("NAME")
        val phoneNo = i.getStringExtra("PHONE_NO")
        summaryString = createSummary(name, phoneNo, hasHotCoffeeCap, hasBrewedCoffeeCap, priceCap, hasHotCoffeeMocha, hasBrewedCoffeeMocha, priceMocha, hasHotCoffeeFrappe, hasBrewedCoffeeFrappe, priceFrappe, hasHotCoffeeEspresso, hasBrewedCoffeeEspresso, priceEspresso, finalPrice)
        displayMessage()
    }

    /**
     * This method displays the Summary in an DialogBox.
     */
    private fun displayMessage() {
        val alertDialog = AlertDialog.Builder(this@MainActivity)
        alertDialog.setTitle(R.string.title)
        alertDialog.setMessage(summaryString)
        alertDialog.setPositiveButton(R.string.okay) { _, _ -> }
        alertDialog.show()
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Cappuccino
     */
    private fun calculatePriceCap(isHotCoffee: Boolean, isBrewedCoffee: Boolean): Int {
        var price = 50
        if (isHotCoffee) {
            price += 5
        }
        if (isBrewedCoffee) {
            price += 7
        }
        return quantityCap * price
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Caffe Mocha
     */
    private fun calculatePriceMocha(isHotCoffee: Boolean, isBrewedCoffee: Boolean): Int {
        var price = 70
        if (isHotCoffee) {
            price += 5
        }
        if (isBrewedCoffee) {
            price += 7
        }
        return quantityMocha * price
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Frappe Coffee
     */
    private fun calculatePriceFrappe(isHotCoffee: Boolean, isBrewedCoffee: Boolean): Int {
        var price = 75
        if (isHotCoffee) {
            price += 5
        }
        if (isBrewedCoffee) {
            price += 7
        }
        return quantityFrappe * price
    }

    /**
     * Calculates the price received from SubmitOrder method
     *
     * @param isHotCoffee    is used to check whether customer needs Hot Coffee
     * @param isBrewedCoffee is used to check whether customer needs Brewed Coffee
     * @return price of Espresso
     */
    private fun calculatePriceEspresso(isHotCoffee: Boolean, isBrewedCoffee: Boolean): Int {
        var price = 90
        if (isHotCoffee) {
            price += 5
        }
        if (isBrewedCoffee) {
            price += 7
        }
        return quantityEspresso * price
    }

    /**
     * This method creates an Summary of Total order by User.
     */
    private fun createSummary(name: String?, phone: String?, hasHotCoffeeCap: Boolean, hasBrewedCoffeeCap: Boolean, pCap: Int, hasHotCoffeeMocha: Boolean, hasBrewedCoffeeMocha: Boolean, pMocha: Int, hasHotCoffeeFrappe: Boolean, hasBrewedCoffeeFrappe: Boolean, pFrappe: Int, hasHotCoffeeEspresso: Boolean, hasBrewedCoffeeEspresso: Boolean, pEspresso: Int, price: Int): String {
        var summary = """
             ${getString(R.string.name_summary)}$name
             ${getString(R.string.phone_no_summary)}$phone
             ${getString(R.string.view_line)}
             """.trimIndent()
        //Cappuccino
        summary += """
             
             ${getString(R.string.cappuccino)}
             """.trimIndent()
        summary += """
             
             ${getString(R.string.hot_coffee_summary)}$hasHotCoffeeCap
             ${getString(R.string.brewed_coffee_summary)}$hasBrewedCoffeeCap
             """.trimIndent()
        summary += """
             
             ${getString(R.string.quantity_summary)}$quantityCap
             ${getString(R.string.price)}$pCap
             """.trimIndent()
        summary += """
             
             ${getString(R.string.view_line)}
             """.trimIndent()
        //Mocha
        summary += """
             
             ${getString(R.string.caffe_mocha)}
             """.trimIndent()
        summary += """
             
             ${getString(R.string.hot_coffee_summary)}$hasHotCoffeeMocha
             ${getString(R.string.brewed_coffee_summary)}$hasBrewedCoffeeMocha
             """.trimIndent()
        summary += """
             
             ${getString(R.string.quantity_summary)}$quantityMocha
             ${getString(R.string.price)}$pMocha
             """.trimIndent()
        summary += """
             
             ${getString(R.string.view_line)}
             """.trimIndent()
        //Frappe
        summary += """
             
             ${getString(R.string.frappe)}
             """.trimIndent()
        summary += """
             
             ${getString(R.string.hot_coffee_summary)}$hasHotCoffeeFrappe
             ${getString(R.string.brewed_coffee_summary)}$hasBrewedCoffeeFrappe
             """.trimIndent()
        summary += """
             
             ${getString(R.string.quantity_summary)}$quantityFrappe
             ${getString(R.string.price)}$pFrappe
             """.trimIndent()
        summary += """
             
             ${getString(R.string.view_line)}
             """.trimIndent()
        //Espresso
        summary += """
             
             ${getString(R.string.espresso)}
             """.trimIndent()
        summary += """
             
             ${getString(R.string.hot_coffee_summary)}$hasHotCoffeeEspresso
             ${getString(R.string.brewed_coffee_summary)}$hasBrewedCoffeeEspresso
             """.trimIndent()
        summary += """
             
             ${getString(R.string.quantity_summary)}$quantityEspresso
             ${getString(R.string.price)}$pEspresso
             """.trimIndent()
        summary += """
             
             ${getString(R.string.view_line)}
             """.trimIndent()
        //Total Price
        summary += """
             
             ${getString(R.string.total_price)}$price
             ${getString(R.string.view_line)}
             """.trimIndent()
        return summary
    }

    /**
     * This method is invoked when CONFIRM ORDER is clicked.
     * It also passes the Summary to Mail using Intent.
     */
    fun mailOrder(view: View?) {
        val subject = getString(R.string.order_subject) + name
        Toast.makeText(this@MainActivity, R.string.thank_you, Toast.LENGTH_LONG).show()
        val intent = Intent(Intent.ACTION_SENDTO)
        intent.data = Uri.parse("mailto:")
        intent.putExtra(Intent.EXTRA_SUBJECT, subject)
        intent.putExtra(Intent.EXTRA_TEXT, summaryString)
        startActivity(intent)
    }
}